renderdoc/replay/entry_points.cpp

```diff
@@ -678,8 +679,318 @@ uint32_t StartAndroidPackageForCapture(const char *host, const char *package)
+{
+  RDCLOG("Checking for existing signature");
+
+  // Get the list of files in META-INF
+  string fileList = execCommand("aapt list " + apk).strStdout;
+  if(fileList.empty())
+    return false;
+
+  // Walk through the output.  If it starts with META-INF, remove it.
+  std::istringstream contents(fileList);
+  string line;
+  string prefix("META-INF");
+  while(std::getline(contents, line))
+  {
+    line = trim(line);
+    if(line.compare(0, prefix.size(), prefix) == 0)
+    {
+      RDCLOG("Match found, removing  %s", line.c_str());
+      execCommand("aapt remove " + apk + " " + line);
+    }
+  }
+
+  // Ensure no hits on second pass through
+  RDCLOG("Walk through file list again, ensure signature removed");
+  fileList = execCommand("aapt list " + apk).strStdout;
+  std::istringstream recheck(fileList);
+  while(std::getline(recheck, line))
+  {
+    if(line.compare(0, prefix.size(), prefix) == 0)
+    {
+      RDCERR("Match found, that means removal failed! %s", line.c_str());
+      return false;
+    }
+  }
+  return true;
+}
+
+bool AddLayerToAPK(const string &apk, const string &layerPath, const string &layerName,
+                   const string &abi, const string &tmpDir)
+{
+  RDCLOG("Adding RenderDoc layer");
+
+  // Create the required directory layout
+  string local("lib/" + abi + "/" + layerName);
+  string global(tmpDir + local);
+  FileIO::CreateParentDirectory(global);
+  FileIO::Copy(layerPath.c_str(), global.c_str(), true);
+
+  Process::ProcessResult result = execCommand("aapt add " + apk + " " + local, tmpDir);
+
+  if(result.strStdout.empty())
+    return false;
+
+  if(!result.strStderror.empty())
+    return false;
+
+  return true;
+}
+
+bool RealignAPK(const string &apk, string &alignedAPK, const string &tmpDir)
+{
+  // Re-align the APK for performance
+  RDCLOG("Realigning APK");
+  string errOut = execCommand("zipalign -f 4 " + apk + " " + alignedAPK, tmpDir).strStderror;
+
+  if(!errOut.empty())
+    return false;
+
+  // Wait until the aligned version exists to proceed
+  uint32_t elapsed = 0;
+  uint32_t timeout = 10000;    // 10 seconds
+  while(elapsed < timeout)
+  {
+    std::ifstream infile(alignedAPK.c_str());
+    if(infile.good())
+    {
+      RDCLOG("Aligned APK ready to go, continuing...");
+      return true;
+    }
+
+    Threading::Sleep(1000);
+    elapsed += 1000;
+  }
+
+  RDCERR("Timeout reached aligning APK");
+  return false;
 }
 
+#if ENABLED(RDOC_WIN32)
+static string debugKey("%USERPROFILE%\\.android\\debug.keystore");
+#else
+static string debugKey("~/.android/debug.keystore");
+#endif
+
+bool DebugSignAPK(const string &apk, const string &workDir)
+{
+  RDCLOG("Signing with debug key");
+
+// clang-format off
+#if ENABLED(RDOC_WIN32)
+  execCommand(
+      "cmd /C apksigner.bat sign "
+      "--ks " + debugKey + " "
+      "--ks-pass pass:android "
+      "--key-pass pass:android "
+      "--ks-key-alias androiddebugkey " +
+      apk + "\\",
+      workDir);
```

Comment:
Yes, apksigner is actually a script, so it needed an interpreter on each operating system.  I can look into hiding the implementation details.

I'll rework the arguments and re-enable clang format.